### PR TITLE
Allow returning keys before the next frame.

### PR DESCRIPTION
**Pull request by dvc94ch**
Superceeds #1103

### CODE REVIEW

File: quinn-proto/src/connection/mod.rs [hunk 1]
```diff
@@ -1569,7 +1569,7 @@ where
                     _ => unreachable!("got updated secrets during 1-RTT"),
                 }
             }
-            if outgoing.is_empty() {
+            if outgoing.is_empty() && space == self.highest_space {
                 break;
             }
```
Comment (Ralith) [inline quinn-proto/src/connection/mod.rs:1574]:
```suggestion
            if outgoing.is_empty() {
                if space == self.highest_space {
                    break;
                } else {
                    // Keys updated, check for another frame
                    continue;
                }
            }
```

**Review by Ralith — CHANGES_REQUESTED** [review]
Thanks! As written I think this will produce unnecessary zero-length CRYPTO frames, so one minor tweak.

**Review by Ralith — APPROVED** [review]

**Comment by dvc94ch** [comment]
One more potential change would be adding an api for specifiying a custom version. Any opinions on what the best way to make sure that quinn-noise uses a different quic version than quic-tls?

**Comment by Ralith** [comment]
We should definitely support that for this case. I think a decent minimal solution would be to add something like `override_version: Option<u32>`, or `Option<RangeInclusive<u32>>`, to `EndpointConfig`. A better solution would make it harder to inadvertently run an endpoint with a standard version but a nonstandard configuration, but nothing immediately occurs to me.

**Comment by dvc94ch** [comment]
But we probably want to support all ietf versions. So maybe we can change the `0xff` version prefix to something else?

**Comment by Ralith** [comment]
I'm not sure I follow. The 0xff prefix in IETF draft versions is part of the version; a non-standard QUIC variant would not be an IETF draft. Quinn doesn't support all IETF draft versions because support for long-obsolete drafts is both difficult and not particularly useful. If you want to integrate breaking post-v1 changes to IETF QUIC into a non-standard QUIC variant, you'd probably do so by allocating an additional non-standard version.

**Pull request merged**